src/App.tsx

```diff
@@ -15,48 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  Default = '',
+  Name = 'name',
+  Lenght = 'length',
+}
+
+function getPreparedGoods(
+  goodsFrom: string[],
+  sortField: SortType,
+  hasReverse: boolean,
+) {
+  const goods = [...goodsFrom];
+
+  if (sortField) {
+    goods.sort((goodA, goodB) => {
+      switch (sortField) {
+        case SortType.Name:
+          return goodA.localeCompare(goodB);
+
+        case SortType.Lenght:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (hasReverse) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+function getResetValue(
+  sortField: SortType,
+  hasReverse: boolean,
+) {
+  if (sortField.length > 0 || hasReverse) {
+    return true;
+  }
+
+  return false;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [hasReverse, setHasReverse] = useState(false);
```

Comment:
* Consider adding a generic type here
* Accordingly to the convention boolean variables for states and props should have the name ```isReversed, setIsReversed```